crates/compiler/mono/src/code_gen_help/equality.rs

```diff
@@ -206,6 +204,10 @@ fn eq_tag_union<'a>(
     }
 
     let body = match union_layout {
+        NonRecursive(&[]) => {
+            // cannot be reached at runtime, but we need to generate valid code
+            Stmt::Ret(Symbol::BOOL_TRUE)
+        }
```

Comment:
I'm not sure why the wasm backend did not run into this. using the normal logic led to an integer overflow (underflow?) in the equality logic.

Reply:
The wasm backend itself has some equality logic. It only asks CodeGenHelp to generate a procedure for structs. If it detects a simple number layout, it just generates a single instruction. Similarly if it detects void or unit layout, it just generates `==` as `i32.const 1`.

https://github.com/roc-lang/roc/blob/6091ccd3b06d5f0c192fc2aedc84b4941e102216/crates/compiler/gen_wasm/src/low_level.rs#L2026-L2030